modules/3d/test/test_odometry.cpp

```diff
@@ -254,45 +275,60 @@ void OdometryTest::run()
     int iterCount = 100;
     int better_1time_count = 0;
     int better_5times_count = 0;
-    for(int iter = 0; iter < iterCount; iter++)
+    for (int iter = 0; iter < iterCount; iter++)
     {
         Mat rvec, tvec;
         generateRandomTransformation(rvec, tvec);
+
         Mat warpedImage, warpedDepth;
         warpFrame(image, depth, rvec, tvec, K, warpedImage, warpedDepth);
         dilateFrame(warpedImage, warpedDepth); // due to inaccuracy after warping
 
-        isComputed = odometry->compute(image, depth, mask, warpedImage, warpedDepth, mask, calcRt);
-        if(!isComputed)
-            continue;
+        OdometryFrame odfSrc = odometry.createOdometryFrame();
+        OdometryFrame odfDst = odometry.createOdometryFrame();
+        odfSrc.setImage(image);
+        odfSrc.setDepth(depth);
+        odfDst.setImage(warpedImage);
+        odfDst.setDepth(warpedDepth);
 
+        odometry.prepareFrames(odfSrc, odfDst);
+        isComputed = odometry.compute(odfSrc, odfDst, calcRt);
+
+        if (!isComputed)
+            continue;
         Mat calcR = calcRt(Rect(0,0,3,3)), calcRvec;
         cv::Rodrigues(calcR, calcRvec);
         calcRvec = calcRvec.reshape(rvec.channels(), rvec.rows);
         Mat calcTvec = calcRt(Rect(3,0,1,3));
-
 #if SHOW_DEBUG_IMAGES
         imshow("image", image);
         imshow("warpedImage", warpedImage);
         Mat resultImage, resultDepth;
         warpFrame(image, depth, calcRvec, calcTvec, K, resultImage, resultDepth);
         imshow("resultImage", resultImage);
-        waitKey();
+        waitKey(100);
 #endif
```

Comment:
BTW, there is 

```
if (cvtest::debugLevel >= 10)
{
    ...
}
```

details: #18955 , use ">= 10" for UI interaction.

Reply:
done